### DISCUSSION TITLE
Must use import to load ES Module

### DISCUSSION THREAD

**Discussion by jaskarannagi19** [babel/babel#11858] [Q&A]
I am new to this babel and with some experience in nodejs. My package.json files has a babel

"@babel/preset-react": "^7.10.4",
    "@nuxtjs/eslint-config": "^0.0.1",
    "babel-cli": "^6.26.0",

These dependencies related to babel. Got this project form some developer who is unable to work. After installing dependencies and other things. When I try to run the project via nuxtjs give me following error

Must use import to load ES Module: /home/jaskaran/Downloads/electrictourcompany.com-master/node_modules/@babel/runtime/helpers/esm/objectSpread2.js require() of ES modules is not supported. require() of /home/jaskaran/Downloads/electrictourcompany.com-master/node_modules/@babel/runtime/helpers/esm/objectSpread2.js from /home/jaskaran/Downloads/electrictourcompany.com-master/node_modules/vue-server-renderer/build.dev.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules. Instead rename objectSpread2.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /home/jaskaran/Downloads/electrictourcompany.com-master/node_modules/@babel/runtime/helpers/esm/package.json. 

I have already tried renaming and removing the type module from the node_module folder which is usually recommend. Now I am lost on moving forward

**Comment by JLHwung** [comment]
Please attach your babel config. Try to search [`useESModules`](https://babeljs.io/docs/en/babel-plugin-transform-runtime#useesmodules) option of `@babel/transform-runtime`, it should be set as `false` because your build tools has generated

```js
require("@babel/runtime/helpers/esm/objectSpread2");
```

And esmodules can not be `require`d.

Unrelated: `babel-cli` is for Babel 6, please remove it and use `@babel/cli` instead.

**Reply by jaskarannagi19** [reply]
Thanks for the reply @JLHwung . Project doesn't the .babelrc config file. I not sure how it is using it some how this error is gone. But node still throws an error when following code is called 


const website = await app.$axios.$post(`/singletons/get/website`, {
    simple: true,
    populate: 3,
    rspc: 1,
    fields: {
      title: 1,
      canonicalURL: 1,
      searchAction: 1,
      organization: 1,
      homepage: 1,
      twitter: 1,
      mainMenu: 1,
      secondaryMenu: 1,
      brand: 1,
      footer: 1,
      footerMenu: 1,
      author: 1,
      map: 1
    }
  })


NuxtServerError
connect ECONNREFUSED 127.0.0.1:80

It's trying to connect to something on localhost which I am not sure to what and the previous dev says nothing about that

**Comment by jaskarannagi19** [comment]
The second thing is when I run npm start I get following error:


/home/jaskaran/Downloads/electrictourcompany.com-master/nuxt.config.js:1
(function (exports, require, module, __filename, __dirname) { import path from 'path'
                                                              ^^^^^^

SyntaxError: Cannot use import statement outside a module
    at new Script (vm.js:88:7)
    at createScript (vm.js:263:10)
    at Object.runInThisContext (vm.js:311:10)
    at wrapSafe (internal/modules/cjs/loader.js:1041:15)
    at Module._compile (internal/modules/cjs/loader.js:1102:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! electrictourcompany@1.0.0 start: `cross-env NODE_ENV=production node server/index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the electrictourcompany@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


I also added .babelrc from babel website as follows:

{
  "presets": ["@babel/preset-env"]
}

**Reply by JLHwung** [reply]
It seems to me that you are running the _uncompiled_ source directly from node.js.

Babel does not magically compile your app. If you are using `@babel/cli`, you should compile `server/` and run the compiled javascript files.

https://babeljs.io/docs/en/babel-cli#compile-directories

```
babel server --out lib
cross-env NODE_ENV=production node lib/index.js
```

**Reply by jaskarannagi19** [reply]
Thanks for the feedback. Well you are correct I did run uncompiled code from the babel but whenever I compiled that single file.

npx babel server --out-dir lib

npx cross-env NODE_ENV=production node lib/index.js
/home/jaskaran/Downloads/electrictourcompany.com/nuxt.config.js:1
(function (exports, require, module, __filename, __dirname) { import path from 'path'
                                                              ^^^^^^

SyntaxError: Cannot use import statement outside a module
    at new Script (vm.js:88:7)
    at createScript (vm.js:263:10)
    at Object.runInThisContext (vm.js:311:10)
    at wrapSafe (internal/modules/cjs/loader.js:1041:15)
    at Module._compile (internal/modules/cjs/loader.js:1102:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
 

I also tried to use nuxt generate it does generate a .nuxt folder which contains a compiled files now I try to run this:
npx babel-node server.js 

(node:35090) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
/home/jaskaran/Downloads/electrictourcompany.com/.nuxt/server.js:1
import { stringify } from 'querystring';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:1054:16)
    at Module._compile (internal/modules/cjs/loader.js:1102:27)
    at loader (/home/jaskaran/Downloads/electrictourcompany.com/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.<computed> [as .js] (/home/jaskaran/Downloads/electrictourcompany.com/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at Object.<anonymous> (/home/jaskaran/Downloads/electrictourcompany.com/node_modules/babel-cli/lib/_babel-node.js:154:22)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)


Error still remains same. NOTE: package.json doesnt contain set "type": "module" nor does the project has any .babelrc file also when I try same code and deploy it to netlify server. It compile successfully Here the log


npm run generate
12:30:58 AM: > electrictourcompany@1.0.0 generate /opt/build/repo
12:30:58 AM: > nuxt generate
12:31:05 AM: [error] (node:1579) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
12:32:49 AM: [error]  /blog
12:32:49 AM: 
12:32:49 AM: Error: Request failed with status code 502
12:32:49 AM:     at createError (/opt/build/repo/node_modules/axios/lib/core/createError.js:16:15)
12:32:49 AM:     at settle (/opt/build/repo/node_modules/axios/lib/core/settle.js:17:12)
12:32:49 AM:     at IncomingMessage.handleStreamEnd (/opt/build/repo/node_modules/axios/lib/adapters/http.js:237:11)
12:32:49 AM:     at IncomingMessage.emit (events.js:203:15)
12:32:49 AM:     at IncomingMessage.EventEmitter.emit (domain.js:448:20)
12:32:49 AM:     at endReadableNT (_stream_readable.js:1145:12)
12:32:49 AM:     at process._tickCallback (internal/process/next_tick.js:63:19)
12:38:13 AM: ​
12:38:13 AM: (build.command completed in 7m 14.9s)
12:38:13 AM: ​
12:38:13 AM: ┌────────────────────────────────────────────────────────────┐
12:38:13 AM: │ 2. onPostBuild command from @netlify/plugin-functions-core │
12:38:13 AM: └────────────────────────────────────────────────────────────┘
12:38:13 AM: ​
12:38:13 AM: Packaging Functions from assets/functions directory:
12:38:13 AM:  - geocode.js
12:38:13 AM:  - instagram.js
12:38:13 AM:  - reverse-geocode.js
12:38:14 AM: ​
12:38:14 AM: (@netlify/plugin-functions-core onPostBuild completed in 1.1s)
12:38:14 AM: ​
12:38:14 AM: ┌─────────────────────────────┐
12:38:14 AM: │   Netlify Build Complete    │
12:38:14 AM: └─────────────────────────────┘
12:38:14 AM: ​
12:38:14 AM: (Netlify Build completed in 7m 17.5s)
12:38:14 AM: Caching artifacts
12:38:14 AM: Started saving node modules
12:38:14 AM: Finished saving node modules
12:38:14 AM: Started saving build plugins
12:38:14 AM: Finished saving build plugins
12:38:14 AM: Started saving pip cache
12:38:14 AM: Finished saving pip cache
12:38:14 AM: Started saving emacs cask dependencies
12:38:14 AM: Finished saving emacs cask dependencies
12:38:14 AM: Started saving maven dependencies
12:38:14 AM: Finished saving maven dependencies
12:38:14 AM: Started saving boot dependencies
12:38:14 AM: Finished saving boot dependencies
12:38:14 AM: Started saving go dependencies
12:38:14 AM: Finished saving go dependencies
12:38:18 AM: Build script success
12:38:18 AM: Starting to deploy site from 'dist'
12:38:20 AM: Creating deploy tree asynchronously
12:38:20 AM: Creating deploy upload records
12:38:22 AM: 437 new files to upload
12:38:22 AM: 3 new functions to upload
12:38:39 AM: Starting post processing
12:38:39 AM: Post processing - HTML
12:39:01 AM: Finished processing build request in 8m55.394651865s
12:39:57 AM: Post processing - redirect rules
12:39:57 AM: Post processing - header rules
12:39:57 AM: Post processing done
12:39:57 AM: Site is live

**Reply by jaskarannagi19** [reply]
I am startinf to have a feeling that its not related to babel if it was this should crash on production server too. Only the dev build is crashing. But production compiles the code before running. I did the same compile the code with nuxt generate command code compile which error ar the end.

 ERROR  Could not resolve routes                                                                                                                                 01:31:42


 FATAL  connect ECONNREFUSED 127.0.0.1:80
